Limit debug console size to current window size.

galegosimpatico opened the pull request:
If any stored preferences for the debug console size happens to be larger than its corresponding dimension at the current window size, ignore the preference and create the debug console _panel_ using the window size values instead of the stored preferences.

_I prepared this one today... originally to be directly pushed to `trunk`. But I'm rather creating a pull request for it because maybe it is better to wait some days or weeks before adding low importance changes._

galegosimpatico committed:
Limit debug console size to current window size.

If any stored preferences for the debug console size happens to be
larger than its corresponding dimension at the current window size,
ignore the preference and create the debug console _panel_ using the
window size values instead of the stored preferences.

ghost commented:
I would be merging this by myself as soon as am sure [the `imgui` upgrade](https://github.com/anura-engine/anura/commit/33edd62522580e8362fbce2a8269d498b6f385ff) is safe... I had some problems on my Debian box after it was applied... so I opened this https://github.com/anura-engine/imgui/pull/2 pull request just in case, but I hope it's my fault.

galegosimpatico committed:
Limit debug console size to current window's (2).

src/debug_console.cpp: Replace ternary operator with more (idiomatic?)
clamp call.
